gvsbuild/outdated.py

```diff
@@ -40,11 +40,16 @@ def outdated():
     try:
         for project in projects.values():
             try:
-                latest_version = lastversion.latest(
-                    repo=project.repository or project.name,
-                    major=project.lastversion_major,
-                    even=project.lastversion_even,
-                )
+                for repo in filter(None, [project.repository, project.name]):
+                    try:
+                        latest_version = lastversion.latest(
+                            repo=repo,
+                            major=project.lastversion_major,
+                            even=project.lastversion_even,
+                        )
+                        break
+                    except lastversion.exceptions.BadProjectError:
```

Comment:
why can't we keep it as before and just add the new except after InvalidVersion?